### PR TITLE
Handling 16 bytes at once

### PR DESCRIPTION
**Pull request by artsiomkorzun**
**NOTE:** This challenge closes for submissions on Jan 31 23:59:59 UTC. No new pull requests for adding submissions will be accepted after that time.
Any pending pull requests will be reviewed over the next few days, as described [here](https://github.com/gunnarmorling/1brc/discussions/687).
The final leader board will be published by Feb 5.

#### Check List:

- [x] You have run `./mvnw verify` and the project builds successfully
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time: -
* Execution time of reference implementation: -

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Nice improvement! Thank you so much for participating in 1BRC and congrats on creating one of the very top entries! You've added so many tricks adopted by others too, way to go!

```
Benchmark 1: timeout -v 300 ./calculate_average_artsiomkorzun.sh 2>&1
  Time (mean ± σ):      1.587 s ±  0.005 s    [User: 0.001 s, System: 0.004 s]
  Range (min … max):    1.579 s …  1.598 s    10 runs

Summary
  artsiomkorzun: trimmed mean 1.587168676875, raw times 1.578604839,1.5904227530000001,1.585700493,1.587893237,1.586567214,1.588402829,1.585967739,1.584054185,1.588340965,1.5981434490000002

Leaderboard

| # | Result (m:s.ms) | Implementation     | JDK | Submitter     | Notes     |
|---|-----------------|--------------------|-----|---------------|-----------|
|   | 00:01.587 | [link](https://github.com/gunnarmorling/1brc/blob/main/src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java)| 21.0.2-graal | [Artsiom Korzun](https://github.com/artsiomkorzun) | GraalVM native binary, uses Unsafe |
```

**Comment by artsiomkorzun** [comment]
@gunnarmorling Thank you very much for the challenge. I definitely learned something new and am eager to apply all these SWAR tricks in the real-life code. @thomaswue thank you for the best ILP loop and hinting that the crucial method is not being inlined. @merykitty and @royvanrijn thank you for the SWAR tricks. @jerrinot thank you for ILP and 16-byte branching. @abeobk thank you for the masking idea which makes the later one work as fast as hell.

That was a fun!

File: src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java [hunk 1]
```diff
@@ -413,26 +415,41 @@ public void run() {
...
+                    long word1 = word(chunk2.position);
+                    long word2 = word(chunk2.position + 8);
+
+                    long separator1 = separator(word1);
                     long separator2 = separator(word2);
-                    long pointer2 = find(aggregates, chunk2, word2, separator2);
-                    long value2 = value(chunk2);
-                    Aggregates.update(pointer2, value2);
+
+                    long pointer = find(aggregates, chunk2, word1, word2, separator1, separator2);
```
Comment (mtopolnik) [inline src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java:436]: Does this approach mean the program segfaults on hash collision?

Comment (artsiomkorzun) [reply src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java:436]: It checks only the first entry, if there is a collision it goes to the bottom. Find is optimistic. Put is pessimistic.

**Comment by gunnarmorling** [comment]
Hey @artsiomkorzun!

Congrats again on being in the Top 3 of the One Billion Row Challenge!

To celebrate this amazing achievement, I would like to send you a 1BRC t-shirt and coffee mug. To claim your prize, fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSeU-gFCzJnWccwUYgMahde5buO6Ny2NQ79isCAjlrGqjZo-Vg/viewform) by Feb 11. **After** submitting the form, please provide a comment with the random value you've specified in the form, so that I know it is you who submitted it.

All data entered will solely be used in relation to processing this shipment. Shipments can be sent to any country listed [here](https://help.spreadshirt.com/hc/de/articles/207165489-Lieferzeit-Versandkosten?locale=de_DE). A big thank you to [Decodable](https://www.decodable.co/) for sponsoring these prizes!

Thanks a lot for participating in 1BRC,

--Gunnar

**Comment by artsiomkorzun** [comment]
Hi @gunnarmorling, thank you very much. The number is 29057 26381

**Pull request merged**